### ISSUE TITLE
Django 1.6 and upcoming 1.7 compat

### ISSUE THREAD

**Issue by xordoquy** [simone/django-compositekey#28]
Hi Simone,

First of all thanks a lot for the project. It has helped me a lot dealing with legacy databases.

I was curious about plans to support Django 1.6 and 1.7. Do you have any ? Do you have pointers about how this could be done ?

**Comment by simone** [comment]
Yes, I think is time to support django 1.6.1...
Actually are you using compositekey with django 1.4 or 1.5?

for 1.7, spep-by-step.

**Comment by xordoquy** [comment]
I'm currently using compositekey with django 1.5. I'm really looking forward to 1.6 for transactions management and 1.7 for the tons of new features :)

**Comment by simone** [comment]
I'm watching 1.6 code, a lot of stuff are changed.
I don't know if is better change approach avoiding monkeypatching library and fork directly django core.

**Comment by xordoquy** [comment]
I have no idea about that.
By the way, did you contact the Django core developers about this work ?

**Comment by epineda** [comment]
Hi Simone!

do you have any news to support django 1.6.x ?

thank you in advance...

**Comment by DebVortex** [comment]
Hi Simone, hi people,

here havent happen anything for a while. Could you please give us a short update what the support about Django 1.6 says?

With kind regards and thanks in advance.

**Comment by simone** [comment]
Hi guys,
you're right. I promis to port it.

I'm always confused on the CompositeKey support from the DSF,
they organised 2 google of summer code, 2011 and 2013 for adding this support.
https://gist.github.com/koniiiik/5408673

but do you know when they will be officially add this in the roadmap?
So let me some weeks and I'll found time to port the code.

**Comment by DebVortex** [comment]
Hi Simone,

I'm only aware of this discussion: https://code.djangoproject.com/wiki/MultipleColumnPrimaryKeys

And this one havent changed for a long time now. At least not significant. I just asked apollo13, one of the CoreDevs if he knows more but he is also not aware of this Topic. He suggested me to ask akaariai in the #django-dev IRC chanel. I will going to do so when i find the spare time and catch him online.

**Comment by simone** [comment]
ok started the development. 
I refactored the unit tests (djangotests/cruntests.py)
now should be more easy found the difference between 1.5 and 1.6

**Comment by ivandeex** [comment]
hi! I made a patch with preliminary support of 1.6 and very beginnings of 1.7. sent directly to your mail.

**Comment by simone** [comment]
Thanks for your help. I fix the test part and fixture serialization issues.
Now the test run.. but too many error for a stable release.

---

Ran 115 tests in 0.737s

FAILED (failures=12, errors=9, skipped=3, expected failures=3, unexpected successes=1)
Destroying test database for alias 'default'...
Destroying test database for alias 'other'...

**Comment by auvipy** [comment]
here is another useful link https://code.djangoproject.com/wiki/CompositePrimaryKeyDesignNotes

I think you should mail to django dev mailing list for current state and implementation design decisions.

**Comment by simone** [comment]
Yes, this discussion is very old. I talked personally with the CORE development, that really don't care the problem. 2 Summer of code has been done for an implementation that is started since django 1.2.
I implemented this library in Django 1.4. Ported to django 1.5 (but seem that there are some troubles on this - I Never use this library with django 1.5).
The django 1.6 changed all the test code base structure, and the ORM now supports multiple foreign keys, that should help the table joins... but not really because the ORM is not written for supports the primary keys.
django 1.7 is another big refactoring. it changed the ORM but there is no this feature,  not yet.
I'd like to find time to rewrite this library for 1.6. But really there are only 2 or 3 person interested. I should beg forgiveness to all of you but I'm going long in this implementation.

thanks

**Comment by auvipy** [comment]
Hi from django mailing list https://groups.google.com/forum/#!topic/django-developers/MZUcOE6-7GY

seems some initiatives are seen to add composite key. interested people might help in the process or share your ideas from technical point of view

**Comment by simone** [comment]
Thanks for sharing

**Comment by facconi** [comment]
Hi Simone, thank you for your work! Do you have any news on updating this app to support the new versions of Django? Thanks!
